src/sbuf/MemBlob.cc

```diff
@@ -72,8 +75,9 @@ MemBlob::~MemBlob()
 {
     if (mem || capacity)
         memFreeBuf(capacity, mem);
-    Stats.liveBytes -= capacity;
-    --Stats.live;
+    auto stats = GetStatsInstance();
```

Comment:
This and similar PR code is buggy because it copies MemBlobStats -- among other "decorations", `auto` does not automatically relay references. We have to explicitly ask for a reference if we want a reference instead of a regular object (unlike, say, pointers, references are not C++ objects).

I am a bit surprised that the compiler cannot detect the following "pointless" `stats` update!

Reply:
Fixed. Doh. :(